Generate data extracts based on the geom type (POINT, LINE, POLYGON) selected.

Sujanadh opened the pull request:
## What type of PR is this? (check all applicable)

- [X] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation
- [ ] 🧑‍💻 Refactor
- [ ] ✅ Test
- [ ] 🤖 Build or CI
- [ ] ❓ Other (please specify)

## Related Issue

Issues: 
- #2302
- #2159

## Describe this PR

- Specify geom type to generate data extracts. 
```python
geom_type = DbGeomType.Polygon
```
as a default parameter in the backend.
- If `osm_buildings` category is selected, then we allow `Polygons`, `Points` and `Points along with the building centroids`
- If `osm_buildings` is selected as a category, the option to select Lines as a primary geom type needs to be hidden, I guess? @spwoodcock , @NSUWAL123 
- By default, the frontend passes centroid: true if point data is selected. Later on the advanced option, we will allow the toggle option for the centroid.

## Screenshots

![image](https://github.com/user-attachments/assets/98854837-5854-4768-a4d4-338d3717ef51)

## Alternative Approaches Considered

Did you attempt any other approaches that are not documented in code?

## Review Guide

Notes for the reviewer. How to test this change?

## Checklist before requesting a review

- 📖 Read the FMTM Contributing Guide: <https://github.com/hotosm/fmtm/blob/main/CONTRIBUTING.md>
- 📖 Read the HOT Code of Conduct: <https://docs.hotosm.org/code-of-conduct>
- 👷‍♀️ Create small PRs. In most cases, this will be possible.
- ✅ Provide tests for your changes.
- 📝 Use descriptive commit messages.
- 📗 Update any related documentation and include any relevant screenshots.

## [optional] What gif best describes this PR or how it makes you feel?

spwoodcock commented on src/frontend/src/components/createnewproject/DataExtract.tsx:
If the user selects 'Points', then it's pretty easy to display a checkbox toggle as you say
- 'Include polygon centroids'
👍

Sujanadh commented:
## Updates:
- removed separate yaml files for points, centroid and polygons
- used PyYaml to adjust the config of buildings.yaml based on the geom type requests
- updated frontend to add checkbox to allow polygons centroid only when Point is selected
![Screenshot from 2025-04-15 22-18-18](https://github.com/user-attachments/assets/59f78b75-0f4d-4556-890f-66061d3004c4)

spwoodcock reviewed (approved):
Nice - good frontend change too 🎉
